pandas/core/arrays/timedeltas.py

```diff
@@ -162,15 +162,30 @@ def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
         result._freq = freq
         return result
 
-    def __new__(cls, values, freq=None, dtype=_TD_DTYPE):
+    def __new__(cls, values, freq=None, dtype=_TD_DTYPE, copy=False):
 
         freq, freq_infer = dtl.maybe_infer_freq(freq)
 
-        values = np.array(values, copy=False)
-        if values.dtype == np.object_:
-            values = array_to_timedelta64(values)
```

Comment:
What is the more or less the difference between `array_to_timedelta64` and `sequence_to_td64ns` ?

Reply:
`sequence_to_td64ns` is the more general case, wraps `array_to_timedelta64`